### PR TITLE
Real Time Clock Drivers

### LINKED ISSUE #8: Real Time Clock (RTC) Drivers
We have a real time clock on the telemetry board, and we need drivers to have a nice interface to use it. Note that we are not using the real time clock built into the stm32f429 (because of an oversight during the hardware design process). Instead we are using the [PCF8523T](https://www.nxp.com/docs/en/data-sheet/PCF8523.pdf) as an external real time clock.

### PR DESCRIPTION
**Pull request by sugamarora99**

### CODE REVIEW

File: Docs/source/drivers.rst [hunk 1]
```diff
@@ -1,2 +1,18 @@
...
+*******
+
+Real-Time Clock (RTC)
+=====================
+
+Purpose
+    The RTC drivers keep track of date and time. The RTC uses ``I2C2``.
+
+Usage
+    First the ``RTC_Init(IMUData_t* Data)`` must be called. It will initialize the peripheral to collect data using the "12-hour mode"
```
Comment (manthanand) [inline Docs/source/drivers.rst]: Should probably change IMUData to something else.

File: Drivers/Periphs/Inc/RTC.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+typedef struct{
+    uint8_t sec; // Will not use these parameters unless we have to
+    uint8_t min; // ^^
+    uint8_t hr; // ^^
+    uint8_t days; // ^^
+    uint8_t weekdays;
+    uint8_t months;
+    uint8_t years;
+} RTCData_t;
+
```
Comment (manthanand) [inline Drivers/Periphs/Inc/RTC.h:39]: Comments here for description of function and @param

File: Drivers/Periphs/Src/RTC.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+RTCData_t *RTCData;
+
+void RTC_Init(RTCData_t *data){
+    //configure RTC
+    uint8_t config[2];
+    config[0] = CONTROL_1; //register address
+    config[1] = 0x08; //data to be transmitted 
+    HAL_I2C_Master_Transmit_IT(&hi2c2, DEVICE_ADDRESS, config, 2); // select 12 hour mode 
+    
+    data = RTCData;
```
Comment (manthanand) [inline Drivers/Periphs/Src/RTC.c]: I think you want to set RTCData to data.

File: Drivers/Periphs/Src/RTC.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+void RTC_UpdateMeasurements(){
+    uint8_t data[7];
+    uint8_t reg_addr = SECONDS;
+    HAL_I2C_Master_Transmit_IT(&hi2c2, DEVICE_ADDRESS, &reg_addr, 1); //send register address to read from
```
Comment (manthanand) [inline Drivers/Periphs/Src/RTC.c]: Should probably include something if these function return an error.

File: Drivers/Periphs/Src/RTC.c (hunk 3 above)
Comment (sugamarora99) [reply Drivers/Periphs/Src/RTC.c]: Fixed

File: Drivers/Periphs/Src/RTC.c (hunk 4 above)
Comment (sugamarora99) [reply Drivers/Periphs/Src/RTC.c]: Fixed

File: Core/Inc/main.h [hunk 5]
```diff
@@ -37,6 +37,15 @@ extern "C" {
 
 /* Exported types ------------------------------------------------------------*/
 /* USER CODE BEGIN ET */
+CAN_HandleTypeDef hcan1;
```
Comment (manthanand) [inline Core/Inc/main.h]: Should leave these in the main.c file and extern them here.

File: Drivers/Periphs/Src/RTC.c [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Sets Real-Time Clock to 12 hour mode (this can be changed to 24 hour mode instead if desired)
+ * @param data buffer that will hold data 
+**/
+ErrorStatus RTC_Init(RTCData_t *data){
+    //configure RTC
+    uint8_t config[2];
+    config[0] = CONTROL_1; //register address
+    config[1] = 0x08; //data to be transmitted 
+    HAL_I2C_Master_Transmit_IT(&hi2c2, DEVICE_ADDRESS, config, 2); // select 12 hour mode
```
Comment (manthanand) [inline Drivers/Periphs/Src/RTC.c]: Shouldn't error be set equal to the output of this function? Same for the rest.

Comment (sugamarora99) [reply Drivers/Periphs/Src/RTC.c]: fixed

File: Core/Inc/main.h (hunk 5 above)
Comment (sugamarora99) [reply Core/Inc/main.h]: Fixed in latest commit

**Review by manthanand — APPROVED** [review]

**Pull request closed without merging**